nullaway/src/test/java/com/uber/nullaway/jspecify/GenericMethodTests.java

```diff
@@ -947,6 +951,93 @@ public void arrayCovariantSubtyping() {
         .doTest();
   }
 
+  @Test
+  public void rowMapperTest() {
+    makeHelperWithInferenceFailureWarning()
+        .addSourceLines(
+            "TestRowMapper.java",
+            "import java.util.List;",
+            "import org.jspecify.annotations.*;",
+            "@NullMarked",
+            "class TestRowMapper {",
+            "",
+            "  interface RowMapper<T extends @Nullable Object> {}",
+            "",
+            "  static class SingleColumnRowMapper<T> implements RowMapper<@Nullable T> {",
+            "    public SingleColumnRowMapper(Class<T> requiredType) {}",
+            "  }",
+            "",
+            "  protected <U> RowMapper<@Nullable U> getSingleColumnRowMapper(Class<U> requiredType) {",
+            "    return new SingleColumnRowMapper<>(requiredType);",
+            "  }  ",
+            "  ",
+            "  public <V extends @Nullable Object> List<V> query(RowMapper<V> rowMapper) {",
+            "    throw new RuntimeException();",
+            "  }",
+            "  ",
+            "  public <W> List<@Nullable W> queryForList(Class<W> elementType) {",
+            "    return query(getSingleColumnRowMapper(elementType));",
+            "  }",
+            "}")
+        .doTest();
+  }
```

Comment:
Test case might benefit from a quick explanation of what's being tested, since no reports are expected. From looking at the test alone, the answer is that we are testing constraint resolution when method invocations are passed as parameters to other methods with some nested type variable on the signature of both methods. But given the PR description, it is a bit more specific than that, no? We are making sure we avoid conflating `T` and `@Nullable T`, right?

Reply:
Yeah, I'll add a comment.  This code was reduced from some code in Spring that was giving weird inference failures, and yes, it was due to not properly distinguishing `T` and `@Nullable T`